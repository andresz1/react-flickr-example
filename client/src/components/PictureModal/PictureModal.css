.container {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.8);
}

.modal {
  position: relative;
  color: white;
}

.content {
  color: white;
  position: absolute;
  bottom: 0;
  width: calc(100% - 16px);
  padding: 8px;
  font-size: 12px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.image {
  max-width: 100%;
  height: auto;
  max-height: 70vh;
}

.btn {
  width: fit-content;
  border-style: none;
  background-color: transparent;
  cursor: pointer;
  margin: 0 8px;
}

.btn i {
  color: white;
  font-size: 32px;
}

.close {
  position: absolute;
  top: 16px;
  right: 16px;
  composes: btn;
}
