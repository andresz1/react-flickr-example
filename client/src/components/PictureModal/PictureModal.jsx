// @flow

import React, { Component }  from 'react';

import styles from './PictureModal.css';

type IProps = {
  picture: IPicture,
  onNext: Function,
  onPrevious: Function,
  onClose: Function,
};

class PictureModal extends Component<IProps> {
  constructor(props: IProps){
    super(props);

    this.handleKeyDown = this.handleKeyDown.bind(this);
  }

  toggleScroll(active: boolean) {
    if (document.documentElement) {
      document.documentElement.style.overflow = (active) ? 'scroll' : 'hidden';
    }
  }

  /*:: handleKeyDown: () => void */
  handleKeyDown(e: KeyboardEvent) {
    const code: number = e.keyCode || e.which;

    if (code === 37) {
      this.props.onPrevious();
    } else if (code === 39) {
      this.props.onNext();
    } else if (code === 27) {
      this.props.onClose();
    }
  }

  componentDidMount() {
    this.toggleScroll(false);

    document.addEventListener('keydown', this.handleKeyDown, false);
  }

  componentWillUnmount() {
    this.toggleScroll(true);

    document.removeEventListener('keydown', this.handleKeyDown, false);
  }

  render() {
    const { picture } = this.props;

    return (
      <div className={styles.container}>
        <button
          className={styles.btn}
          onClick={this.props.onPrevious}
        >
          <i className="material-icons">
            chevron_left
          </i>
        </button>

        <div className={styles.modal}>
          <img
            className={styles.image}
            src={picture.src}
            alt=""
          />
        </div>

        <div className={styles.content}>
          Title: {picture.title}
          <br/>
          Tags: {picture.tags}
        </div>

        <button
          className={styles.btn}
          onClick={this.props.onNext}>
          <i className="material-icons">
            chevron_right
          </i>
        </button>

        <button
          className={styles.close}
          onClick={this.props.onClose}>
          <i className="material-icons">
            close
          </i>
        </button>
      </div>
    );
  }
}

export default PictureModal;
