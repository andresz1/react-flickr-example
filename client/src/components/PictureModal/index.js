import PictureModal from './PictureModal.jsx';

export default PictureModal;
