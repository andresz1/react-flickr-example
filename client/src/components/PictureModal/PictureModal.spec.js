import React from 'react';
import { shallow } from 'enzyme';
import PictureModal from './PictureModal';

const picture = {
  id: '44592167171',
  owner: '166588675@N05',
  secret: '144e4f8561',
  server: '1848',
  farm: 2,
  title: 'Olden 3',
  ispublic: 1,
  isfriend: 0,
  isfamily: 0,
  ownername: 'OldenCam3',
  tags: '',
  src: 'https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg',
  href: 'https://www.flickr.com/photos/166588675@N05/44592167171/'
};

const setup = (setupProps = {}) => {
  const defaultProps = {
    picture,
    handleNext: jest.fn(),
    handlePrevious: jest.fn(),
    handleClose: jest.fn(),
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <PictureModal
      picture={props.picture}
      onNext={props.handleNext}
      onPrevious={props.handlePrevious}
      onClose={props.handleClose}
    />
  );

  return {
    props,
    wrapper
  };
};

describe('<PictureModal />', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onPrevious when clicked', () => {
    const { props, wrapper } = setup();

    wrapper.find('button').first().simulate('click');
    expect(props.handlePrevious).toHaveBeenCalled();
  });

  it('calls onNext when clicked', () => {
    const { props, wrapper } = setup();

    wrapper.find('button').at(1).simulate('click');
    expect(props.handleNext).toHaveBeenCalled();
  });

  it('calls onClose when clicked', () => {
    const { props, wrapper } = setup();

    wrapper.find('button').last().simulate('click');
    expect(props.handleClose).toHaveBeenCalled();
  });

  it('calls onPrevious when left key is pressed', () => {
    const KEYBOARD_LEFT_CODE = 37;
    const events = {
      keydown: null,
    };

    document.addEventListener = jest.fn((event, cb) => {
      events[event] = cb;
    });

    const { props } = setup();

    events.keydown({ keyCode: KEYBOARD_LEFT_CODE });

    expect(props.handlePrevious).toHaveBeenCalled();
  });

  it('calls onNext when right key is pressed', () => {
    const KEYBOARD_RIGHT_CODE = 39;
    const events = {
      keydown: null,
    };

    document.addEventListener = jest.fn((event, cb) => {
      events[event] = cb;
    });

    const { props } = setup();

    events.keydown({ keyCode: KEYBOARD_RIGHT_CODE });

    expect(props.handleNext).toHaveBeenCalled();
  });

  it('calls onClose when esc key is pressed', () => {
    const KEYBOARD_ESCAPE_CODE = 27;
    const events = {
      keydown: null,
    };

    document.addEventListener = jest.fn((event, cb) => {
      events[event] = cb;
    });

    const { props } = setup();

    events.keydown({ keyCode: KEYBOARD_ESCAPE_CODE });

    expect(props.handleClose).toHaveBeenCalled();
  });
});
