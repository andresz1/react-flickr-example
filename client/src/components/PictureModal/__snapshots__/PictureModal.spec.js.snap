// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PictureModal /> renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PictureModal
    onClose={[Function]}
    onNext={[Function]}
    onPrevious={[Function]}
    picture={
        Object {
            "farm": 2,
            "href": "https://www.flickr.com/photos/166588675@N05/44592167171/",
            "id": "44592167171",
            "isfamily": 0,
            "isfriend": 0,
            "ispublic": 1,
            "owner": "166588675@N05",
            "ownername": "OldenCam3",
            "secret": "144e4f8561",
            "server": "1848",
            "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
            "tags": "",
            "title": "Olden 3",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn"
          onClick={[Function]}
>
          <i
                    className="material-icons"
          >
                    chevron_left
          </i>
</button>,
        <div
          className="modal"
>
          <img
                    alt=""
                    className="image"
                    src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
          />
</div>,
        <div
          className="content"
>
          Title: 
          Olden 3
          <br />
          Tags: 
          
</div>,
        <button
          className="btn"
          onClick={[Function]}
>
          <i
                    className="material-icons"
          >
                    chevron_right
          </i>
</button>,
        <button
          className="close"
          onClick={[Function]}
>
          <i
                    className="material-icons"
          >
                    close
          </i>
</button>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="material-icons"
>
            chevron_left
</i>,
          "className": "btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "chevron_left",
            "className": "material-icons",
          },
          "ref": null,
          "rendered": "chevron_left",
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            className="image"
            src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
/>,
          "className": "modal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "image",
            "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Title: ",
            "Olden 3",
            <br />,
            "Tags: ",
            "",
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          "Title: ",
          "Olden 3",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "Tags: ",
          "",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="material-icons"
>
            chevron_right
</i>,
          "className": "btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "chevron_right",
            "className": "material-icons",
          },
          "ref": null,
          "rendered": "chevron_right",
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="material-icons"
>
            close
</i>,
          "className": "close",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "material-icons",
          },
          "ref": null,
          "rendered": "close",
          "type": "i",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn"
            onClick={[Function]}
>
            <i
                        className="material-icons"
            >
                        chevron_left
            </i>
</button>,
          <div
            className="modal"
>
            <img
                        alt=""
                        className="image"
                        src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
            />
</div>,
          <div
            className="content"
>
            Title: 
            Olden 3
            <br />
            Tags: 
            
</div>,
          <button
            className="btn"
            onClick={[Function]}
>
            <i
                        className="material-icons"
            >
                        chevron_right
            </i>
</button>,
          <button
            className="close"
            onClick={[Function]}
>
            <i
                        className="material-icons"
            >
                        close
            </i>
</button>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="material-icons"
>
              chevron_left
</i>,
            "className": "btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "chevron_left",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "chevron_left",
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              className="image"
              src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
/>,
            "className": "modal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "image",
              "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              "Olden 3",
              <br />,
              "Tags: ",
              "",
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            "Olden 3",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Tags: ",
            "",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="material-icons"
>
              chevron_right
</i>,
            "className": "btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "chevron_right",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "chevron_right",
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="material-icons"
>
              close
</i>,
            "className": "close",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "close",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "close",
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
